usmqe/api/graphiteapi/graphiteapi.py

```diff
@@ -34,4 +35,7 @@ def get_datapoints(self, target, from_date=None, until_date=None):
         response = requests.get(
             pytest.config.getini("graphite_api_url") + pattern)
         self.check_response(response)
-        return response.json()[0]["datapoints"]
+        try:
+            return response.json()[0]["datapoints"]
+        except IndexError as err:
+            return response.json()
```

Comment:
I don't read the doc for requests nor I try it. But are you sure that if `IndexError` happens `response.json()` is empty list as the method description says?

Reply:
I have tested graphite api and with specified `format=json` it returns empty list `[]` when the `target` for `render` API call doesn't exist. Otherwise it returns list of dictionaries with attributes `target` (provided in uri) and `datapoints`.
I have encountered that the list of dictionaries have only one item but it would be probably good to check length of it.